presto-hudi/src/main/java/com/facebook/presto/hudi/HudiSplitManager.java

```diff
@@ -145,17 +146,22 @@ private ExtendedFileSystem getFileSystem(ConnectorSession session, HudiTableHand
 
     private HudiSplit createHudiSplit(HudiTableHandle table, FileSlice slice, String timestamp, HudiPartition partition)
     {
-        HoodieBaseFile hoodieBaseFile = slice.getBaseFile().get();
-        HudiFile baseFile = new HudiFile(hoodieBaseFile.getPath(), 0, hoodieBaseFile.getFileLen());
+
+        HudiFile hudiFile = slice.getBaseFile().map(hoodieBaseFile -> new HudiFile(hoodieBaseFile.getPath(), 0, hoodieBaseFile.getFileLen()))
+                .orElseGet(() ->
+                    table.getTableType()  == HudiTableType.MOR ?
+                            slice.getLogFiles().findFirst().map(hoodieLogFile -> new HudiFile(hoodieLogFile.getPath().toString(), 0, hoodieLogFile.getFileSize())).orElse(null)
+                            : null
+                );
+
         List<HudiFile> logFiles = slice.getLogFiles()
                 .map(logFile -> new HudiFile(logFile.getPath().toString(), 0, logFile.getFileSize()))
                 .collect(toImmutableList());
-
         return new HudiSplit(
                 table,
                 timestamp,
                 partition,
-                Optional.of(baseFile),
+                Optional.ofNullable(hudiFile),
```

Comment:
If the split have no base file, we should not put a log file into HudiSplit#baseFile field. This could add misleading  when we will create a new MOR reader to handle the file slice that has no base file.  Just put `hudFile` I suggested above.

Reply:
For now, we could take use of the first log file as base file in HudiRecordCursors.createRealtimeRecordCursor.